2_Scope_Closures/practice/closure_part1_yujo.js

```diff
@@ -0,0 +1,49 @@
+const isPrime = (function isPrime() {
+  const map = { 1: false, 2: true, 3: true };
+
+  function checkPrime(v) {
+    if (map[v] === undefined) {
+      if (v % 2 == 0 || v % 3 == 0) {
+        return (map[v] = false);
+      }
+      var vSqrt = Math.sqrt(v);
+      for (let i = 5; i <= vSqrt; i += 6) {
+        if (v % i == 0 || v % (i + 2) == 0) {
+          return (map[v] = false);
+        }
+      }
+      return (map[v] = true);
+    }
+    return map[v];
+  }
```

Comment:
```suggestion
function checkPrime(v) {
  if (map[v]) {
    return map[v];
  }
  if (v % 2 == 0 || v % 3 == 0) {
    return (map[v] = false);
  }
  var vSqrt = Math.sqrt(v);
  for (let i = 5; i <= vSqrt; i += 6) {
    if (v % i == 0 || v % (i + 2) == 0) {
      return (map[v] = false);
    }
  }
  return (map[v] = true);
}
```
해당 로직은 map[v] 가 있을 경우, 빠르게 리턴을 시켜주는 게 가독성과 성능 면에서 더 효율성을 높일 수 있지 않을까 싶습니다 ㅎㅎ